StripePayments/StripePayments/Source/Helpers/STPBankAccountCollector.swift

```diff
@@ -370,59 +370,57 @@ public class STPBankAccountCollector: NSObject {
         onEvent: ((FinancialConnectionsEvent) -> Void)?,
         completion: @escaping STPCollectBankAccountForSetupCompletionBlock
     ) {
-        guard let setupIntentID = STPSetupIntent.id(fromClientSecret: clientSecret) else {
-            completion(nil, error(for: .invalidClientSecret))
+        let setupIntentID = STPSetupIntent.id(fromClientSecret: clientSecret)
+        logCollectBankAccountStarted(type: .setup, intentID: setupIntentID)
+        // Overwrite completion to send an analytic before calling the caller-supplied completion
+        let completion: (FinancialConnectionsSDKResult?, STPSetupIntent?, NSError?) -> Void = { result, setupIntent, error in
+            self.logCollectBankAccountFinished(type: .setup, intentID: setupIntent?.stripeID, linkAccountSessionID: nil, financialConnectionsSDKResult: result, error: error)
+            completion(setupIntent, error)
+        }
+        guard let setupIntentID else {
+            completion(nil, nil, error(for: .invalidClientSecret))
             return
         }
         let financialConnectionsCompletion:
             (FinancialConnectionsSDKResult?, LinkAccountSession?, NSError?) -> Void = {
                 result,
                 linkAccountSession,
                 error in
-                if let error = error {
-                    completion(
-                        nil,
-                        self.error(for: .unexpectedError, userInfo: [NSUnderlyingErrorKey: error])
-                    )
+                if let error {
+                    completion(result, nil, error as NSError)
                     return
                 }
-                guard let linkAccountSession = linkAccountSession,
-                    let result = result
-                else {
-                    completion(nil, NSError.stp_genericFailedToParseResponseError())
+                guard let result else {
+                    completion(result, nil, self.error(for: .unexpectedError, loggingSafeErrorMessage: "Completed without a result"))
```

Comment:
To be consistent, change to `collectBankAccountForSetup() completed without a result`

Reply:
Thank you!